vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthSplashCarouselFragment.kt

```diff
@@ -80,17 +80,27 @@ class FtueAuthSplashCarouselFragment @Inject constructor(
                     "Branch: ${BuildConfig.GIT_BRANCH_NAME}"
             views.loginSplashVersion.debouncedClicks { navigator.openDebug(requireContext()) }
         }
+        views.splashCarousel.registerAutomaticUntilInteractionTransitions()
+    }
+
+    private fun ViewPager2.registerAutomaticUntilInteractionTransitions() {
+        var scheduledTransition: Job? = null
+        registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
+            private var hasUserManuallyInteractedWithCarousel: Boolean = false
 
-        views.splashCarousel.apply {
-            var scheduledTransition: Job? = null
-            registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
-                override fun onPageSelected(position: Int) {
-                    scheduledTransition?.cancel()
+            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
+                hasUserManuallyInteractedWithCarousel = !isFakeDragging
+            }
+
+            override fun onPageSelected(position: Int) {
+                scheduledTransition?.cancel()
+                if (hasUserManuallyInteractedWithCarousel) {
+                    // stop the automatic transitions
```

Comment:
This is fine but maybe avoid having empty block. I think lint/compiler will also complain. Better to write `if(!...)` and just add the comment above.

Reply:
in this case the comment counts as a non empty block, will invert the condition and remove the branch

Reply:
01ef46517e0358d0446a9b13fd0612e6286c4883